src/software/thunderscope/play/playinfo_widget.py

```diff
@@ -0,0 +1,42 @@
+import pyqtgraph as pg
+import pyqtgraph.console as pg_console
+from proto.play_info_msg_pb2 import PlayInfo
+from software.networking.threaded_unix_listener import ThreadedUnixListener
+import software.thunderscope.constants as constants
+import queue
+
+from proto.robot_log_msg_pb2 import RobotLog
+
+
+class playInfoWidget(pg.DataTreeWidget):
+    def __init__(self, buffer_size=10):
+        pg.DataTreeWidget.__init__(self)
+
+        self.log_buffer = queue.Queue(buffer_size)
+        
+    def refresh(self):
+        """Update the play info widget with new robot information
+        """
+        try:
+            playinfo = self.log_buffer.get_nowait()
+        except queue.Empty as empty:
+            return
+
+        d = {
+            'play name': playinfo.play.play_name,
+            'robot 0': {'tactic name' : playinfo.robot_tactic_assignment[0].tactic_name,
+                        'tactic fsm state' : playinfo.robot_tactic_assignment[0].tactic_fsm_state},
+            'robot 1': {'tactic name' : playinfo.robot_tactic_assignment[1].tactic_name,
+                        'tactic fsm state' : playinfo.robot_tactic_assignment[1].tactic_fsm_state},
+            'robot 2': {'tactic name' : playinfo.robot_tactic_assignment[2].tactic_name,
+                        'tactic fsm state' : playinfo.robot_tactic_assignment[2].tactic_fsm_state},
+            'robot 3': {'tactic name' : playinfo.robot_tactic_assignment[3].tactic_name,
+                        'tactic fsm state' : playinfo.robot_tactic_assignment[3].tactic_fsm_state},
+            'robot 4': {'tactic name' : playinfo.robot_tactic_assignment[4].tactic_name,
+                        'tactic fsm state' : playinfo.robot_tactic_assignment[4].tactic_fsm_state},
+            'robot 5': {'tactic name' : playinfo.robot_tactic_assignment[5].tactic_name,
+                        'tactic fsm state' : playinfo.robot_tactic_assignment[5].tactic_fsm_state}           
+        }
```

Comment:
we need to make this slightly dynamic because we won't always have all 5 robots

checkout this PR I opened here: https://github.com/alinas2000/Software/pull/1

It uses MessageToDict to convert the protobuf to a dictionary, and uses the table widget instead of the tree widget which seems to be flatter/not have the length=2 thing that the data tree widget adds

If you can agree with the changes in the PR you can merge it into your branch